### PR TITLE
docs: added a guide for native messaging implementors

### PR DESCRIPTION
**Pull request by shakyShane**
To read the docs in the rendered format

```
npm run docs && open docs/index.html
```

Screenshot 2023-07-21 at 11 41 24

### CODE REVIEW

**Comment by shakyShane** [comment]
Current dependencies on/for this PR:
* main
  * **PR #646** <a href="https://app.graphite.dev/github/pr/duckduckgo/content-scope-scripts/646" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/content-scope-scripts/646?utm_source=stack-comment).

File: packages/messaging/native.js [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+ *
+ * For example, in Android this would be what happens within a `@Javascript` Interface, but on macOS it would be within
+ * the WebKit messaging protocol, etc.
+ *
+ * ## Algorithm
+ *
+ * 1. let `s` be an incoming raw `JSON` payload
+ * 2. let `msg` be the result of parsing `s` into key/value pairs
+ *     - 2.1 Note: 'parsing' here may not be required if the platform in question receives JSON data directly (ie: JavaScript environments)
+ * 3. if parsing was not successful, throw an "invalid message" Exception
```
Comment (muodov) [inline packages/messaging/native.js:31]: Are this and the subsequent exceptions delivered back to the sender?

**Review by muodov — DISMISSED** [review]

File: packages/messaging/native.js [hunk 2]
```diff
@@ -0,0 +1,210 @@
+/**
```
Comment (jonathanKingston) [inline packages/messaging/native.js:1]: Is JSDoc incapable of consuming markdown or similar? This isn't really a JS file.

Comment (shakyShane) [reply packages/messaging/native.js:1]: @jonathanKingston you can use markdown, but then you can't do internal links, which are pretty useful here.

File: packages/messaging/native.js (hunk 1 above)
Comment (shakyShane) [reply packages/messaging/native.js:31]: I'll think about how to best explain this, in a follow-up :)

**Review by jonathanKingston — APPROVED** [review]

**Pull request merged**